### PR TITLE
Improve System.Runtime test coverage

### PR DESCRIPTION
**Pull request by jkotas**
Contributes to https://github.com/dotnet/coreclr/issues/12782

### CODE REVIEW

**Comment by jkotas** [comment]
Complements https://github.com/dotnet/coreclr/pull/20532

File: src/System.Runtime/tests/System/TypeLoadExceptionTests.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        [Fact]
+        public static void Ctor_String_Exception()
+        {
+            string message = "type failed to load";
+            var innerException = new Exception("Inner exception");
+            var exception = new TypeLoadException(message, innerException);
+            ExceptionUtility.ValidateExceptionProperties(exception, hResult: COR_E_TYPELOAD, innerException: innerException, message: message);
+            Assert.Equal("", exception.TypeName);
+        }
+    }
```
Comment (danmosemsft) [inline src/System.Runtime/tests/System/TypeLoadExceptionTests.cs]: I am not sure whether there is a test verifying TypeName etc when the runtime calls the private ctor. At least I didn't see any case of `Assert.Throws<TypeLoadException>` that seemed to assign it to something to inspect. So my look was very brief..

Comment (jkotas) [reply src/System.Runtime/tests/System/TypeLoadExceptionTests.cs]: I am focusing just on making sure that there is equivalent CoreFX test coverage for the tests that I am deleting in CoreCLR.

I agree that we can have better coverage for the runtime type loader error handling. The tests that verify exceptions thrown by runtime type loader would belong to CoreCLR. CoreFX does not have the right infrastructure to produce invalid binaries that trigger them.

Opened: https://github.com/dotnet/coreclr/issues/20536

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Pull request merged**